src/generator/index.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2020-present, Elastic NV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+import { execSync } from 'child_process';
+import { existsSync } from 'fs';
+import { mkdir, readFile, writeFile } from 'fs/promises';
+import { bold, cyan } from 'kleur/colors';
+import { join, relative, dirname, basename } from 'path';
+import { progress, write as stdWrite } from '../helpers';
+import { getPackageManager, runCommand } from './utils';
+
+// Templates that are required for setting up new
+// synthetics project
+const templateDir = join(__dirname, '..', '..', 'templates');
+
+export class Generator {
+  pkgManager = 'npm';
+  constructor(public projectDir: string) {}
+
+  async directory() {
+    if (!existsSync(this.projectDir)) {
+      await mkdir(this.projectDir);
+    }
+  }
+
+  async files() {
+    const fileMap = new Map<string, string>();
+    // Setup Synthetics config file
+    const configFile = 'synthetics.config.ts';
+    fileMap.set(
+      configFile,
+      await readFile(join(templateDir, configFile), 'utf-8')
+    );
+
+    // Setup example journey file
+    const journeyFile = 'journeys/todos.journey.ts';
+    fileMap.set(
+      journeyFile,
+      await readFile(join(templateDir, journeyFile), 'utf-8')
```

Comment:
IMO we should actually have a simpler journey file as the first example. Maybe accessing elastic's website or something along those lines.

I found the current example a bit too convoluted.

Reply:
So the main reason for not touching any external systems is that It generates robots traffic with Synthetics user agent and kind of being their pain when it comes to analytics. I would still to a simpler todo mvc page or any page but not a company website.

Reply:
How would you feel about using `example.com`, which was [built by IANA](https://www.iana.org/domains/reserved) exactly for this purpose?

I was thinking of something along the lines of:

```js
import { journey, step, monitor, expect } from '@elastic/synthetics';

journey('Check Example Website Title', ({ page, params }) => {
  // Only relevant for the push command to create
  // monitors in Kibana
  monitor.use({
    id: 'example-monitor',
    schedule: 10,
  });
  
  step('launch website', async () => {
    await page.goto("https://example.com");
  });

  step('assert title', async () => {
    const header = await page.$('h1');
    expect(await header.textContent()).toBe('Example Domain');
  });
});
```

If not doing that then I think just the file below (with no references to TodoMVC) would be enough:

```js
import { journey, step, monitor, expect } from '@elastic/synthetics';

journey('My Example Journey', ({ page, params }) => {
  // Only relevant for the push command to create
  // monitors in Kibana
  monitor.use({
    id: 'example',
    schedule: 10,
  });
  
  step('launch application', async () => {
    await page.goto(params.url);
  });

  step('assert title', async () => {
    const header = await page.$('h1');
    expect(await header.textContent()).toBe('My Webpage');
  });
});
```

Reply:
I do like the example.com idea, But in our docs we always refer to TodoMVC so I am bit skeptical here. @andrewvc WDYT about using example.com for the scaffolding?

Reply:
What about https://elastic.github.io/synthetics-demo/ which is a hosted version of TODO MVC we already have? Example.com is a bit basic.

Reply:
That was the one which the PR was based on https://github.com/elastic/synthetics/blob/da32e68b484938cfd3f001200ca3c819e4464794/templates/journeys/todos.journey.ts and Lucas suggested to move to example.com

Reply:
I do think we should remove the whole embedded webserver business, that was my doing but just confusing in the end.